webapp/tbapi/tbapi/models/tbapp.py

```diff
@@ -0,0 +1,79 @@
+from sqlalchemy import Column
+from sqlalchemy import (
+    Integer, String, Text
+)
+from sqlalchemy.orm import Session
+
+import json
+from werkzeug.security import generate_password_hash, check_password_hash
+
+from .base import db, SerializeMixin
+
+class Note(SerializeMixin, db.Model):
+    __tablename__ = 'notes'
+
+    @classmethod
+    def get_by_patient_id(cls, patient_id):
+      notes = cls.query.filter_by(patient_id=patient_id).all()
+      if notes:
+        return notes
+      else: #empty
+       return notes   
+
+    @classmethod
+    def get_by_note(cls, noteList):
+        notes = Note.filter_by_fields(noteList)
+        if notes:
+            return notes
+        else: #empty
+            return notes
+
+    @classmethod
+    def put(cls, noteList):
+        notes = Note.filter_by_fields(noteList)
+        if notes:
+            return notes    
+        noteDict = Note.load_fields(noteList)        
+        notes = cls(
+            patient_id=noteDict['patient_id'], 
+            text=noteDict['text'], 
+            created=noteDict['created'], 
+            author_id=noteDict['author_id'], 
+            lastmod=noteDict['lastmod'], 
+            flag_type=noteDict['flag_type'])
+        try:
+            db.session.add(notes)
+            db.session.commit()
+        except:
+            db.session.rollback()
+            raise
+        return Note.filter_by_fields(noteList)
+
+    @classmethod
+    def post(cls, noteList):
+        return Note.put(noteList)
+
+    @classmethod
+    def filter_by_fields(cls, noteList):
+        noteDict = Note.load_fields(noteList)
+        return cls.query.filter_by(
+            patient_id=noteDict['patient_id'], 
+            text=noteDict['text'], 
+            created=noteDict['created'], 
+            author_id=noteDict['author_id'], 
+            lastmod=noteDict['lastmod'], 
+            flag_type=noteDict['flag_type']).all()     
+
+
+    @classmethod
+    def load_fields(cls, noteList):
+        jsonObj = json.loads(noteList)
```

Comment:
flask should automatically decode JSON into an object if you pass the `Content-type: application/json` header and access it at `request.json`. Either way, I don't think you need to re-cast all the fields. If there's an unexpected type it'll get caught when the DB tries to save it

[see example](https://stackoverflow.com/questions/20001229/how-to-get-posted-json-in-flask)

Reply:
Can we discuss?